### PR TITLE
Concourse add affinity

### PR DESCRIPTION
**Pull request by pfeodrippe**
Kubernetes now needs podAffinityTerm, added it to worker stateful set

### CODE REVIEW

**Review by gregoryguillou — APPROVED** [review]
Now works on Kubernetes 1.7.1. It was not as described in #1306

File: stable/concourse/templates/worker-statefulset.yaml [hunk 1]
```diff
@@ -28,7 +28,9 @@ spec:
                       "values": ["{{ template "worker.fullname" . }}"]
                     }]
                   },
-                  "topologyKey": "kubernetes.io/hostname"
+                  "podAffinityTerm": {
+                    "topologyKey": "kubernetes.io/hostname"
```
Comment (unguiculus) [inline stable/concourse/templates/worker-statefulset.yaml:32]: How about removing the deprecated annotation as was done e. g. here:
https://github.com/kubernetes/charts/pull/1326/files

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by pfeodrippe** [comment]
@unguiculus gonna do this until this weekend, thanks for pointing this

**Comment by unguiculus** [comment]
Marking this as stale. Please update PR within one week.

File: stable/concourse/templates/worker-statefulset.yaml (hunk 1 above)
Comment (deduktion) [reply stable/concourse/templates/worker-statefulset.yaml:32]: like this?
```apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  name: {{ template "worker.fullname" . }}
  labels:
    app: {{ template "worker.fullname" . }}
    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
    release: "{{ .Release.Name }}"
    heritage: "{{ .Release.Service }}"
spec:
  serviceName: {{ template "worker.fullname" . }}
  replicas: {{ .Values.worker.replicas }}
  template:
    metadata:
      labels:
        app: {{ template "worker.fullname" . }}
      annotations:
        affinity:
          podAntiAffinity:
            preferredDuringSchedulingIgnoredDuringExecution:
              - weight: 100
                labelSelector:
                  matchExpressions:
                    - key: app
                      operator: In
                      values:
                        - {{ template "worker.fullname" . }}
                podAffinityTerm:
                  topologyKey: kubernetes.io/hostname   
    spec:
      terminationGracePeriodSeconds: 60
      containers:
        - name: {{ template "worker.fullname" . }}
          image: "{{ .Values.image }}:{{ .Values.imageTag }}"
          imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
          args:
            - "worker"
          lifecycle:
            preStop:
              exec:
                command:
                  - "concourse"
                  - "land-worker"
                  - "--name=${HOSTNAME}"
          env:
            - name: CONCOURSE_TSA_HOST
              valueFrom:
                configMapKeyRef:
                  name: {{ template "concourse.fullname" . }}
                  key: concourse-tsa-host
            - name: CONCOURSE_TSA_PORT
              valueFrom:
                configMapKeyRef:
                  name: {{ template "concourse.fullname" . }}
                  key: concourse-tsa-port
            - name: CONCOURSE_GARDEN_DOCKER_REGISTRY
              valueFrom:
                configMapKeyRef:
                  name: {{ template "concourse.fullname" . }}
                  key: garden-docker-registry
            - name: CONCOURSE_GARDEN_INSECURE_DOCKER_REGISTRY
              valueFrom:
                configMapKeyRef:
                  name: {{ template "concourse.fullname" . }}
                  key: garden-insecure-docker-registry
            - name: CONCOURSE_TSA_PUBLIC_KEY
              value: "/concourse-keys/host_key.pub"
            - name: CONCOURSE_TSA_WORKER_PRIVATE_KEY
              value: "/concourse-keys/worker_key"
            - name: CONCOURSE_WORK_DIR
              value: "/concourse-work-dir"
          resources: {{ toYaml .Values.worker.resources | indent 12 }}
          securityContext:
            privileged: true
          volumeMounts:
            - name: concourse-keys
              mountPath: /concourse-keys
              readOnly: true
            - name: concourse-work-dir
              mountPath: /concourse-work-dir
      volumes:
        - name: concourse-keys
          secret:
            secretName: {{ template "concourse.fullname" . }}
            defaultMode: 0400
            items:
              - key: host-key-pub
                path: host_key.pub
              - key: worker-key
                path: worker_key
              - key: worker-key-pub
                path: worker_key.pub
  {{- if .Values.persistence.enabled }}
  volumeClaimTemplates:
    - metadata:
        name: concourse-work-dir
        annotations:
          {{- if .Values.persistence.worker.class }}
          volume.beta.kubernetes.io/storage-class: {{ .Values.persistence.worker.class | quote }}
          {{- else }}
          volume.alpha.kubernetes.io/storage-class: default
          {{- end }}
      spec:
        accessModes:
          - {{ .Values.persistence.worker.accessMode | quote }}
        resources:
          requests:
            storage: {{ .Values.persistence.worker.size | quote }}
  {{- else }}
        - name: concourse-work-dir
          emptyDir: {}
  {{- end }}
```

Comment (unguiculus) [reply stable/concourse/templates/worker-statefulset.yaml:32]: yes

**Comment by erickt** [comment]
@unguiculus: I just submitted a PR that implements your suggestion in #1756.

**Pull request closed without merging**